contracts/utils/Votes.sol

```diff
@@ -0,0 +1,210 @@
+     */
+    event DelegateVotesChanged(address indexed delegate, uint256 previousBalance, uint256 newBalance);
+
+    /**
+     * @dev Returns total amount of votes for account.
+     */
+    function getVotes(address account) public view returns (uint256) {
+        return _userCheckpoints[account].latest();
+    }
+
+    /**
+     * @dev Returns total amount of votes at given position.
+     */
+    function _getPastVotes(address account, uint256 timestamp) internal view returns (uint256) {
+        return _userCheckpoints[account].past(timestamp);
+    }
+
+    /**
+     * @dev Retrieve the `totalVotingPower` at the end of `blockNumber`. Note, this value is the sum of all balances.
+     * It is but NOT the sum of all the delegated votes!
+     *
+     * Requirements:
+     *
+     * - `blockNumber` must have been already mined
+     */
+    function getPastTotalSupply(uint256 blockNumber) public view returns (uint256) {
+        require(blockNumber < block.number, "ERC721Votes: block not yet mined");
+        return _totalCheckpoints.past(blockNumber);
+    }
+
+    /**
+     * @dev Get checkpoint for `account` for specific position.
+     */
+    function _getTotalAccountVotesAt(address account, uint32 pos)
+        internal
+        view
+        returns (Checkpoints.Checkpoint memory)
+    {
+        return _userCheckpoints[account].at(pos);
+    }
+
+    /**
+     * @dev Returns total amount of votes.
+     */
+    function _getTotalVotes() internal view returns (uint256) {
+        return _totalCheckpoints.latest();
+    }
+
+    /**
+     * @dev Get number of checkpoints for `account` including delegation.
+     */
+    function _getTotalAccountVotes(address account) internal view returns (uint256) {
+        return _userCheckpoints[account].length();
+    }
+
+    /**
+     * @dev Delegate votes from the sender to `delegatee`.
+     */
+    function delegate(address delegatee) public virtual {
+        address delegator = _msgSender();
+        _delegate(delegator, delegatee, _getDelegatorVotes(delegator));
+    }
+
+    /**
+     * @dev Returns account delegation.
+     */
+    function delegates(address account) public view returns (address) {
+        return _delegation[account];
+    }
+
+    /**
+     * @dev Change delegation for `delegator` to `delegatee`.
+     *
+     * Emits events {DelegateChanged} and {DelegateVotesChanged}.
+     */
+    function _delegate(
+        address account,
+        address newDelegation,
+        uint256 balance
+    ) internal {
+        address oldDelegation = delegates(account);
+        _delegation[account] = newDelegation;
+
+        emit DelegateChanged(account, oldDelegation, newDelegation);
+
+        _moveVotingPower(oldDelegation, newDelegation, balance);
+    }
+
+    /**
+     * @dev Moves voting power.
+     */
+    function _moveVotingPower(
+        address src,
+        address dst,
```

Comment:
Let's rename this to `from` and `to` which are more common throughout the codebase.